tests/models/mplug_owl/test_modeling_mplug_owl.py

```diff
@@ -0,0 +1,458 @@
+        )
+        mask_shape = (input_ids.shape[0], input_ids.shape[1] - 1)
+        inputs_dict = {
+            "input_ids": input_ids.long(),
+            "labels": input_ids.clone(),
+            "pixel_values": None,
+            "num_images": torch.zeros(self.batch_size, device=input_ids.device).long(),
+            "non_padding_mask": torch.ones(*mask_shape, device=input_ids.device).long(),
+            "non_media_mask": torch.ones(*mask_shape, device=input_ids.device).long(),
+            "prompt_mask": torch.ones(*mask_shape, device=input_ids.device).long(),
+            "attention_mask": torch.ones(*(mask_shape[0], mask_shape[1] + 1), device=input_ids.device).long(),
+        }
+        return config, inputs_dict
+
+    def prepare_config_and_inputs_for_common(self):
+        config, inputs_dict = self.prepare_config_and_inputs()
+        return config, inputs_dict
+
+    def create_and_check_decoder_model_past_large_inputs(self, config, inputs_dict):
+        model = MplugOwlForConditionalGeneration(config=config).language_model.to(torch_device).eval()
+        input_ids = inputs_dict["input_ids"]
+        attention_mask = inputs_dict["attention_mask"]
+
+        # first forward pass
+        outputs = model(input_ids, attention_mask=attention_mask, use_cache=True)
+
+        output, past_key_values = outputs.to_tuple()
+
+        # create hypothetical multiple next token and extent to next_input_ids
+        next_tokens = ids_tensor((self.batch_size, 3), config.vocab_size)
+        next_attn_mask = ids_tensor((self.batch_size, 3), 2)
+
+        # append to next input_ids and
+        next_input_ids = torch.cat([input_ids, next_tokens], dim=-1)
+        next_attention_mask = torch.cat([attention_mask, next_attn_mask], dim=-1)
+
+        output_from_no_past = model(next_input_ids, attention_mask=next_attention_mask, output_hidden_states=True)[
+            "hidden_states"
+        ][-1]
+        output_from_past = model(
+            next_tokens, attention_mask=next_attention_mask, past_key_values=past_key_values, output_hidden_states=True
+        )["hidden_states"][-1]
+
+        # select random slice
+        random_slice_idx = ids_tensor((1,), output_from_past.shape[-1]).item()
+        output_from_no_past_slice = output_from_no_past[:, -3:, random_slice_idx].detach()
+        output_from_past_slice = output_from_past[:, :, random_slice_idx].detach()
+
+        self.parent.assertTrue(output_from_past_slice.shape[1] == next_tokens.shape[1])
+
+        # test that outputs are equal for slice
+        self.parent.assertTrue(torch.allclose(output_from_past_slice, output_from_no_past_slice, atol=1e-2))
+
+
+@require_torch
+class MplugOwlModelTest(ModelTesterMixin, unittest.TestCase):
+    all_model_classes = (MplugOwlForConditionalGeneration,) if is_torch_available() else ()
+    all_generative_model_classes = (MplugOwlForConditionalGeneration,) if is_torch_available() else ()
+    is_encoder_decoder = True
+    test_pruning = False
```

Comment:
If pruning is not tested / not needed, you could remove all code related with pruning on the modeling file